### ISSUE TITLE
test coverage should be increased

### ISSUE THREAD

**Issue by dankelley** [ArgoCanada/argoFloats#253]
Now that I have the travisCI working, it's time to start looking at the code coverage.  I'm going to assign this to @j-harbin.  If you look at e.g. https://codecov.io/gh/ArgoCanada/argoFloats/src/develop/R/subset.R you'll see that quite a bit of the code is not being exercised by the tests.  Red highlighting means not tested.

We ought to increase the test coverage, because that will help us to find bugs.  For example, at line 278 of that file, you can see a big chunk of code for subset by column is highlighted red, i.e. is not tested by the test suite or by examples.  That file has 57.19% coverage.

Here's what I suggest:

1. Take note of that percentage.
2. Add a test for subset-by-column for a built-in `.nc` file (therefore, no need to download data for the test).  Of course, the test must pass before you commit.
3. Commit.
4. Visit the above-named site after 30 minutes or so, and take note of the percentage.  It should have increased.

Keep repeating steps like 1 to 4 for other things in this file.  Then move on to other files.  It's not actually a terribly slow thing to do.  (You can even do it locally, without committing, although I've not done that in a while.). It's 10 minutes to write a test, 20 seconds to git-push, then do other work for half an hour and go back to check on the progress.  Obviously, you can do a bunch of tests at once; each one is likely only 5 lines of code in a test-suite file.

The benefit of this code-coverage is that we don't have to wait for a user to tell us that they got a bug while they were trying to get their work done for a deadline.

**Comment by dankelley** [comment]
Below is a screenshot showing how to use locally.  If you click on a filename, it shows the line-by-line details

<img width="1068" alt="Screen Shot 2020-09-25 at 6 51 47 AM" src="https://user-images.githubusercontent.com/99469/94253407-ac594780-fefb-11ea-8583-96add5228b7e.png">

**Comment by j-harbin** [comment]
Thanks for this, Dan. As we discussed in a vftf I have been *loving* doing this detective work. 

As we discussed,  it is not possible to test `getProfiles`, and we will not be testing debug messages, as they are subject to be frequently changed. It should also be noted that `test_plot.R` has a lower % coverage because bathymetry cannot be tested, and that `test_mapApp` is not tested as I'm not sure if it can be.

Further testing has been completed in commit df954ef703b1a679ae845257a9fafe2cb0a54a84 of the develop branch. If you are satisfied with that, please feel free to close this issue. If not, make any further suggestions here.

**Comment by dankelley** [comment]
Looking at the
```R
library(covr)
report()
```
output, I see that our coverage of `plot()` could be improved.  In "develop" commit c7724c9ab1984a873d4d7995286ff24fb9b04351, I have added the ability to use  `oce::get.topo()` objects instead of just `marmap::getNOAAbathy()` objects.  That means that we can test on both contoured and imaged bathymetry files.

To see how to use this bathy type, see "Example 5B" in the docs for `plot,argoFloats-method()`.

@j-harbin maybe you could add both contour and image testing to our test suite?  That would get quite a lot more of the `plot()` method tested and reported in our codecov reports.

Thanks!

**Comment by j-harbin** [comment]
Done in commit 5cccda542b6894c4ea5be852db28c333b3c95169 of the develop branch.

**Comment by dankelley** [comment]
Thanks.  Now we have as below (from `covr::report()`, run locally).  I see that `misc.R` is the least-covered, so let's think about that a bit ... looking in more detail, the problem is that there's no coverage of the `useAdjustedProfile()` function.  But ... hmmm ... we don't actually use that code anywhere in the package (see `git grep useAdjustedProfile` output).  I think that's a remnant of code I wrote here as a first draft, but then moved over into oce.  @j-harbin, have we ever said in a video that users ought to write `argoFloats::useAdjustedProfile` on their data?  If not, I think this block of code ought to be commented out for a few weeks (as a precaution) and then deleted say at the end of October.  What say you, @j-harbin?

<img width="607" alt="Screen Shot 2020-09-29 at 8 17 32 AM" src="https://user-images.githubusercontent.com/99469/94552132-c9a85180-022c-11eb-869c-0f0ac0552129.png">

**Comment by j-harbin** [comment]
I agree with this. While I was working with codecov I came across this `useAdjustedProfile` and didn't look into the code closely enough. I assumed it was a function used with `useAdjusted`. I will comment this code out.

**Comment by dankelley** [comment]
Not that it matters much, but I was curious as to how `covr` computes the overall score.  I copied the output from the report (snapshotted 2 comments above) and code it as follows:
```R
data <- "
R/AllClass.R    434 95  74  21  56  77.890
R/debug.R       110 62  45  17  41  72.58
R/download.R    95  48  25  23  16  52.08
R/get.R         516 188 96  92  4   51.06
R/mapApp.R      844 538 0   538 0   0.00
R/merge.R       73  33  17  16  1   51.52
R/misc.R        209 78  15  63  1   19.23
R/plot.R        657 315 197 118 2   62.54
R/qc.R          259 75  55  20  1   73.33
R/read.R        204 102 91  11  23  89.22
R/show.R        42  23  16  7   1   69.57
R/subset.R      624 290 232 58  7   80.00
R/summary.R     72  37  26  11  1   70.27
"
d <- read.table(text=data,
   col.names=c("file","lines","relevant","covered","missed","hitsPerLine","coverage"))
message("covr::report() says that 44.85% of the code is covered")
message("My avg1: ", round(100*mean(d$covered)/mean(d$relevant),2), "% covered")
message("My avg2: ", round(100*mean(d$covered/d$relevant),2), "% covered")
message("My avg3: ", round(100*sum(d$covered)/sum(d$relevant),2), "% covered")
```
with results
```
covr::report() says that 44.85% of the code is covered
My avg1: 47.19% covered
My avg2: 59.18% covered
My avg3: 47.19% covered
```
so I'm not sure what `covr` is doing.  Possibly I wrecked the data in my copy/paste, but I sort of doubt that.

**Comment by j-harbin** [comment]
Interesting. This website may be useful, particularly the " How is coverage calculated? " part https://docs.codecov.io/docs/frequently-asked-questions

It says it is done using the following formula:

round((hits / (hits + partials + misses)) * 100, 5) = Coverage Ratio

**Comment by dankelley** [comment]
Oh, interesting.  I don't see 'partials' listed in the `covr::report()` output.

**Comment by dankelley** [comment]
I'm going to go ahead and close this, since we have increased test coverage, so the issue (as defined by its title) has been addressed successfully.  Of course, we will continue to add tests, but there's no reason to keep the issue open as a reminder.  (Put another way, there is basically no endpoint except 100% coverage, but some of the uncovered things would be for code that is self-evidently correct, e.g. raising an error if a file cannot be found.)
